lib/cdo/script_constants.rb

```diff
@@ -352,13 +352,52 @@ def self.assignable_info(course_or_script)
     }
   end
 
+  CSF_COURSE_PATTERNS = [/^(course[a-f])-([0-9]+)$/, /^(express)-([0-9]+)$/, /^(pre-express)-([0-9]+)$/]
+
   def self.has_congrats_page?(script)
     script == ACCELERATED_NAME ||
       ScriptConstants.script_in_category?(:csf_international, script) ||
-      ScriptConstants.script_in_category?(:csf, script) ||
-      ScriptConstants.script_in_category?(:csf_2018, script) ||
-      ScriptConstants.script_in_category?(:csf_2019, script) ||
-      ScriptConstants.script_in_category?(:csf_2020, script)
+      CSF_COURSE_PATTERNS.map {|r| r =~ script}.any?
+  end
+
+  def self.csf_next_course_recommendation(course_name)
+    # These course names without years in them should be mapped statically to their recommendation.
+    static_mapping = {
+      "course1" => "course2",
+      "course2" => "course3",
+      "course3" => "course4",
+      "accelerated" => "course4",
+      "course4" => "applab"
+    }
+
+    return static_mapping[course_name] if static_mapping.include?(course_name)
+
+    # For CSF courses with years in their name, separate into prefix and year. Determine the recommended
+    # next prefix based on constant mapping, then add the year to the recommended prefix.
+    # Example: coursea-2019 becomes prefix: coursea, year: 2019.
+    # coursea maps to courseb, so return courseb-2019.
+    CSF_COURSE_PATTERNS.each do |r|
+      match_data = r.match(course_name)
+      next unless match_data
+
+      prefix = match_data[1]
+      year = match_data[2]
+
+      return "applab" if %w(coursef express).include?(prefix)
+
+      prefix_mapping = {
+        "coursea" => "courseb",
+        "courseb" => "coursec",
+        "coursec" => "coursed",
+        "coursed" => "coursee",
+        "coursee" => "coursef",
+        "pre-express" => "coursec"
```

Comment:
Do we really recommend pre-express to course C?

Reply:
i would think it would be pre-express to express

Reply:
That's what it was before 🤷‍♀️ Should I change it?

Reply:
Especially given the timing, I would say that it's ok to leave any config the way you found it. I wouldn't want to have to roll this back because of an unwanted behavior change. but @dmcavoy if you think this is incorrect, now could be a good time to fix it while we're thinking about it. Do you think this is worth bugging Mike H about?

Reply:
I have no idea how this is used right now but it might be worth just asking quickly while we are thinking about it and fixing for the future.